src/nc_monitor.c

```diff
@@ -0,0 +1,149 @@
+/*
+ * twemproxy - A fast and lightweight proxy for memcached protocol.
+ * 
+ * Copyright (C) 2021, wei huang <wei.kukey@gmail.com>
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <nc_monitor.h>
+#include <nc_rbtree.h>
+
+struct rbtree monitor_tree;
+struct rbnode monitor_sentinel_node;
+
+void monitor_init()
```

Comment:
This seems to be a singleton, but twemproxy has support for multiple `server_pool`s. Instead, it would make more sense to embed a different monitor

It would be counterproductive for debugging and waste time if you got responses that were sent to a pool that wasn't the one you were debugging, and also leak data to applications.
- you should get data from all servers in the pool the client is connected to, but not in servers of other pools

All of these should accept a `struct monitor_data *monitor`, and the actual data itself should be embedded in the server pool?

Reply:
It seems embedded in the server pool is more better and more useful for users.Thanks!

Reply:
In addition, by comparison I think it is better to use struct mhdr (STALQ) for monitor client conn list. Because the `struct array` del elem is not easy.